x-pack/test/performance/utils.ts

```diff
@@ -5,8 +5,26 @@
  * 2.0.
  */
 
+import { Page } from 'playwright';
+
 export function serializeApmGlobalLabels(obj: any) {
   return Object.entries(obj)
     .filter(([, v]) => !!v)
     .reduce((acc, [k, v]) => (acc ? `${acc},${k}=${v}` : `${k}=${v}`), '');
 }
+
+export async function waitForVisualizations(page: Page, visCount: number) {
```

Comment:
Should this raise an exception after a (configurable) timeout? This would fail the build instead of leaving it hanging?

Reply:
Exactly ! However I believe better solution for this to use [Playwright waitForFunction function](https://github.com/elastic/kibana/pull/140074/files/da2e1e39b68e3febc7ff9a9d91790ca2c0b81a01#diff-3e597ae5ce370e61e30420f6f68b0bdaaa91db6ffc33dd5b7c0529f6515658afL48) which has both timeout mechanism and different frequency settings (default one is animation frame). As I mentioned in PR description depending on solution we would decide (If we can disable CSP pros/cons of it) I'd either delete this completely or update with timeout/frequency logic

Reply:
I'm in favor of disabling CSP, but only if we can confirm with @elastic/kibana-security that by doing so we will just be opening up the ability to write better tests, and that we won't be impacting the performance characteristics of Kibana in a serious way.

Additionally, @elastic/kibana-security, do you know if we'll be able to disable CSP on cloud instances? I'm sure that we will want to execute these tests against cloud instances at some point.

Reply:
Thanks for the ping!

> but only if we can confirm with https://github.com/orgs/elastic/teams/kibana-security that by doing so we will just be opening up the ability to write better tests, and that we won't be impacting the performance characteristics of Kibana in a serious way.

I don't think disabling\enabling CSP would impact Kibana performance in any way.

> Additionally, https://github.com/orgs/elastic/teams/kibana-security, do you know if we'll be able to disable CSP on cloud instances? I'm sure that we will want to execute these tests against cloud instances at some point.

If you just need to set `csp.strict` (and potentially `csp.warnLegacyBrowsers`) to `false` to make Kibana play well with `bypassCSP`, then it should be possible to do for a Cloud deployment as well (I see these settings in the ESS allow-list).

**But**, do we know what part of the default CSP rules is interfering with Playwright exactly? Maybe we can just relax CSP a bit with additional directives in `csp.script_src` (or other CSP directives listed [here](https://www.elastic.co/guide/en/kibana/current/settings.html))?

Reply:
> But, do we know what part of the default CSP rules is interfering with Playwright exactly? Maybe we can just relax CSP a bit with additional directives in csp.script_src (or other CSP directives listed [here](https://www.elastic.co/guide/en/kibana/current/settings.html))?

This is certainly an option for a targeted subset of our functional tests, but I want to make sure that we don't relax the CSP for _all_ functional tests. IMO, we need our test suites to verify that code changes don't violate our default CSP.

Reply:
Passing `--csp.strict=false` is working fine for performance journeys and allows us to use [waitForFunction](https://playwright.dev/docs/api/class-page#page-wait-for-function). The only nit is that we get some alertbox during journeys 
<img width="1394" alt="Screenshot 2022-09-07 at 14 56 24" src="https://user-images.githubusercontent.com/9674241/188873189-f54c3423-7016-4d53-bade-fac5cb779fe9.png">
which is not super important imho.

Reply:
> The only nit is that we get some alertbox during journeys

This should do the trick:
```yml
csp.warnLegacyBrowsers: false
```

https://github.com/elastic/kibana/blob/203d26e899a8e3a218be098cb4a4c606b839b613/docs/setup/settings.asciidoc#L76-L80

Reply:
> This is certainly an option for a targeted subset of our functional tests, but I want to make sure that we don't relax the CSP for all functional tests

@legrego we're currently only working with performance tests, which aren't designed to test features, just use the product in some way to get it to produce telemetry.